src/Test/DatabaseResetter.php

```diff
@@ -37,72 +34,52 @@ public static function isDAMADoctrineTestBundleEnabled(): bool
     public static function resetDatabase(KernelInterface $kernel): void
     {
         $application = self::createApplication($kernel);
-        $registry = $kernel->getContainer()->get('doctrine');
+        $databaseResetter = new ORMDatabaseResetter($application, $kernel->getContainer()->get('doctrine'));
 
-        foreach (self::connectionsToReset($registry) as $connection) {
-            $dropParams = ['--connection' => $connection, '--force' => true];
+        $databaseResetter->resetDatabase();
 
-            if ('sqlite' !== $registry->getConnection($connection)->getDatabasePlatform()->getName()) {
-                // sqlite does not support "--if-exists" (ref: https://github.com/doctrine/dbal/pull/2402)
-                $dropParams['--if-exists'] = true;
-            }
-
-            self::runCommand($application, 'doctrine:database:drop', $dropParams);
-
-            self::runCommand($application, 'doctrine:database:create', [
-                '--connection' => $connection,
-            ]);
-        }
-
-        self::createSchema($application, $registry);
+        self::bootFoundry($kernel);
 
         self::$hasBeenReset = true;
     }
 
     public static function resetSchema(KernelInterface $kernel): void
     {
-        $application = self::createApplication($kernel);
-        $registry = $kernel->getContainer()->get('doctrine');
+        foreach (self::schemaResetters($kernel) as $databaseResetter) {
+            $databaseResetter->resetSchema();
+        }
+
+        if (self::isDAMADoctrineTestBundleEnabled()) {
+            return;
+        }
```

Comment:
This is where we have a problem with odm & orm & dama: if we have a global state for odm, it wouldn't be persisted.